pkg/processor/graph/disjoint_set.go

```diff
@@ -0,0 +1,79 @@
+package graph
+
+import "log"
+
+func (g *graph) disjointSubgraphs() []graph {
+	// Init unionFind structure
+	uf := unionFind{
+		parent: make(map[string]string),
+		size:   make(map[string]int),
+		count:  len(g.vertices),
+	}
+	for v := range g.vertices {
+		uf.parent[v] = v
+		uf.size[v] = 1
+	}
+
+	// Find disjoint sets (by uniting connected vertices)
+	for p, vertex := range g.vertices {
+		for _, q := range vertex.edges() {
+			uf.union(p, q)
+		}
+	}
+	log.Printf("Disjoint sets: %v", uf)
+
+	// Build disjoint sets
+	disjointSets := make(map[string][]string)
+	for v := range g.vertices {
+		root := uf.find(v)
+		disjointSets[root] = append(disjointSets[root], v)
+	}
+
+	// Create corresponding subgraphs
+	subgraphs := make([]graph, uf.count, uf.count)
```

Comment:
If length and capacity is the same then `make([]graph, uf.count, uf.count)` is the same as `make([]graph, uf.count)`. Here and if few other places.